stdlib/atexit.pyi

```diff
@@ -1,7 +1,9 @@
-from typing import Any, Callable
+from typing import Any, Callable, TypeVar
+
+_C = TypeVar("_C", bound=Callable[..., Any])
 
 def _clear() -> None: ...
 def _ncallbacks() -> int: ...
 def _run_exitfuncs() -> None: ...
-def register(func: Callable[..., Any], *args: Any, **kwargs: Any) -> Callable[..., Any]: ...
+def register(func: _C, *args: Any, **kwargs: Any) -> _C: ...
```

Comment:
```suggestion
def register(func: Callable[_P, _T], *args: _P.args, **kwargs: _P.kwargs) -> Callable[_P, _T]: ...
```
I'd suggest this (where _P is a ParamSpec and _T an unconstrained TypeVar). This give us a precise return type and precise types for args and kwargs. It doesn't inform the type system that the exact same callable that was passed in will be returned, but it seems more important that we type *args and **kwargs precisely.